### ISSUE TITLE
[BUG] Project system ran into an unexpected problem

### ISSUE THREAD

**Issue by mwkcoding** [microsoft/vscode-dotnettools#1470]
### Describe the Issue

![image](https://github.com/user-attachments/assets/4c7da9a5-965c-4533-be6b-bbbedea3ce05)
[ProjectSystemServerFault_3500c59e-65a9-47b2-8922-9e02834f98f4.failure.txt](https://github.com/user-attachments/files/17010617/ProjectSystemServerFault_3500c59e-65a9-47b2-8922-9e02834f98f4.failure.txt)

### Steps To Reproduce

I'm not sure, it comes up every time I open my repo.

### Expected Behavior

No notification?

### Environment Information

Version: 1.93.1 (system setup)
Commit: 38c31bc77e0dd6ae88a4e9cc93428cc27a56ba40
Date: 2024-09-11T17:20:05.685Z
Electron: 30.4.0
ElectronBuildId: 10073054
Chromium: 124.0.6367.243
Node.js: 20.15.1
V8: 12.4.254.20-electron.0
OS: Windows_NT x64 10.0.22631
Extension: v1.10.18

**Comment by NathanVG** [comment]
I am having the same issue, error happens as soon as I open a project. 

```
=====================
9/17/2024 9:50:48 AM
LimitedFunctionality
System.AggregateException: Project system data flow 'Workspace update handler 56128182' closed because of an exception: Object reference not set to an instance of an object.. (Object reference not set to an instance of an object.)
---> (Inner Exception #0) StreamJsonRpc.RemoteInvocationException: Object reference not set to an instance of an object.
   at StreamJsonRpc.JsonRpc.InvokeCoreAsync[TResult](RequestId id, String targetName, IReadOnlyList`1 arguments, IReadOnlyList`1 positionalArgumentDeclaredTypes, IReadOnlyDictionary`2 namedArgumentDeclaredTypes, CancellationToken cancellationToken, Boolean isParameterObject)
   at Microsoft.VisualStudio.ProjectSystem.VSCode.LanguageServices.Handlers.DynamicItemHandler.<>c__DisplayClass6_0.<<HandleProjectChangeAsync>g__ProcessAdditionsAsync|4>d.MoveNext()
--- End of stack trace from previous location ---
   at Microsoft.VisualStudio.ProjectSystem.VSCode.LanguageServices.Handlers.DynamicItemHandler.HandleProjectChangeAsync(IWorkspaceProject context, IComparable version, IProjectChangeDescription projectChange, ContextState state, ITelemetryOperation telemetryOperation, IManagedProjectDiagnosticOutputService logger)
   at Microsoft.VisualStudio.ProjectSystem.VSCode.LanguageServices.Handlers.DynamicItemHandler.HandleAsync(IWorkspaceProject context, IComparable version, IImmutableDictionary`2 projectChanges, ContextState state, ITelemetryOperation telemetryOperation, IManagedProjectDiagnosticOutputService logger)
   at Microsoft.VisualStudio.ProjectSystem.VSCode.LanguageServices.Workspace.<>c__DisplayClass43_1.<<OnEvaluationUpdateAsync>g__ProcessSourceItemsHandlersAsync|6>d.MoveNext()
--- End of stack trace from previous location ---
   at Microsoft.VisualStudio.ProjectSystem.VSCode.LanguageServices.Workspace.<>c__DisplayClass43_0.<<OnEvaluationUpdateAsync>g__ApplyProjectEvaluationAsync|2>d.MoveNext()
--- End of stack trace from previous location ---
   at Microsoft.VisualStudio.ProjectSystem.VSCode.LanguageServices.Workspace.<>c__DisplayClass46_1`1.<<OnProjectChangedAsync>g__ApplyInBatchAsync|1>d.MoveNext()
--- End of stack trace from previous location ---
   at Microsoft.VisualStudio.Threading.ReentrantSemaphore.StackSemaphore.<>c__DisplayClass3_0.<<ExecuteAsync>b__0>d.MoveNext()
--- End of stack trace from previous location ---
   at Microsoft.VisualStudio.Threading.ReentrantSemaphore.StackSemaphore.ExecuteAsync(Func`1 operation, CancellationToken cancellationToken)
   at Microsoft.VisualStudio.ProjectSystem.OnceInitializedOnceDisposedUnderLockAsync.ExecuteUnderLockCoreAsync(Func`2 action, CancellationToken cancellationToken)
   at Microsoft.VisualStudio.ProjectSystem.VSCode.LanguageServices.Workspace.OnEvaluationUpdateAsync(IProjectVersionedValue`1 evaluationUpdate)
   at Microsoft.VisualStudio.ProjectSystem.VSCode.LanguageServices.Workspace.<>c__DisplayClass42_0.<<OnWorkspaceUpdateAsync>g__ApplyUpdateWithinLockAsync|1>d.MoveNext()
--- End of stack trace from previous location ---
   at Microsoft.VisualStudio.ProjectSystem.VSCode.LanguageServices.Workspace.<>c__DisplayClass42_0.<<OnWorkspaceUpdateAsync>g__ApplyUpdateWithinLockAsync|1>d.MoveNext()
--- End of stack trace from previous location ---
   at Microsoft.VisualStudio.ProjectSystem.VSCode.LanguageServices.Workspace.<>c__DisplayClass42_0.<<OnWorkspaceUpdateAsync>b__0>d.MoveNext()
--- End of stack trace from previous location ---
   at Microsoft.VisualStudio.Threading.ReentrantSemaphore.StackSemaphore.<>c__DisplayClass3_0.<<ExecuteAsync>b__0>d.MoveNext()
--- End of stack trace from previous location ---
   at Microsoft.VisualStudio.Threading.ReentrantSemaphore.StackSemaphore.ExecuteAsync(Func`1 operation, CancellationToken cancellationToken)
   at Microsoft.VisualStudio.ProjectSystem.OnceInitializedOnceDisposedUnderLockAsync.ExecuteUnderLockCoreAsync(Func`2 action, CancellationToken cancellationToken)
   at Microsoft.VisualStudio.ProjectSystem.DataReceivingBlockSlim`1.ProcessInputQueueAsync(InputDataBuffer inputQueue)
<--- (Inner Exception #0) 
===================
```

**Comment by mwkcoding** [comment]
It might be causing the C#/.NET IntelliSense to not work properly. I have to restart vscode 2-5 times until the IntelliSense works properly.

**Comment by dibarbet** [comment]
Two requests
1.  Could you post the `C#` output window logs?
2.  If you're willing, are you able to share the solution you're using?  My email is `dabarbet@microsoft.com` if you'd prefer to share that way.

@tmeschter is there any way the project system side could log the details of the inner `StreamJsonRpc.RemoteInvocation` exception?  That would hopefully give us a stack trace from the Roslyn side.

**Comment by mwkcoding** [comment]
C# log:
```
Using dotnet configured on PATH
Dotnet path: C:\Program Files\dotnet\dotnet.exe
Activating C# + C# Dev Kit + C# IntelliCode...
waiting for named pipe information from server...
[stdout] {"pipeName":"\\\\.\\pipe\\29c20836"}
received named pipe information from server
attempting to connect client to server...
client has connected to server
[Info  - 9:06:46 AM] [Program] Language server initialized
[Info  - 9:07:18 AM] [WorkspaceProjectFactoryService] Project [REDACTED].csproj loaded by C# Dev Kit
... (same message goes for each individual project)
```

I'm not sure I can share my solution. But I can tell you it's a solution containing ~46 projects (WebJobs, Functions, Web Apps and a bunch of shared classes)

**Comment by smitpatel** [comment]
@dibarbet - The extension logs contain details about JSON RPC errors. It is hard to say which file is but we have all logs, we can look through it and identify. It would contain which RPC call failed and whole stacktrace and error.

@mwkcoding - Please share content of extension logs folder for C# dev kit. You can find extension logs folder using command palette in VS Code. Once you open that you can see a folder with name ending in csdevkit. Content of that folder would be useful. You can send the zip for the folder to @dibarbet via email they provided above if you don't comfortable uploading here. Since this folder contains logs for whole session, please have a fresh session of VS Code to capture the logs when error happens.

**Comment by mwkcoding** [comment]
@smitpatel @dibarbet I sent my logs folder zipped to the email provided.

**Comment by dibarbet** [comment]
Thanks got the logs!

@smitpatel I'm seeing a few different issues - it would be useful if you could identify what brokered service and process is being called for each.  It looks like one of the null refs might be in the project system side?  

### StreamJsonRpc.ConnectionLostException
Numerous connection lost exceptions, these are the stack traces I saw
```
StreamJsonRpc.ConnectionLostException: The JSON-RPC connection with the remote party was lost before the request could complete.
   at StreamJsonRpc.JsonRpc.InvokeCoreAsync(JsonRpcRequest request, Type expectedResultType, CancellationToken cancellationToken)
   at StreamJsonRpc.JsonRpc.InvokeCoreAsync[TResult](RequestId id, String targetName, IReadOnlyList`1 arguments, IReadOnlyList`1 positionalArgumentDeclaredTypes, IReadOnlyDictionary`2 namedArgumentDeclaredTypes, CancellationToken cancellationToken, Boolean isParameterObject)
   at Microsoft.VisualStudio.ProjectSystem.VSCode.SolutionPathProvider.GetSolutionPathAsync(CancellationToken cancellationToken)
   at Microsoft.VisualStudio.ProjectSystem.VSCode.LanguageServices.Workspace.<>c__DisplayClass43_0.<<OnEvaluationUpdateAsync>g__ProcessInitialEvaluationDataAsync|1>d.MoveNext()
--- End of stack trace from previous location ---
   at Microsoft.VisualStudio.ProjectSystem.VSCode.LanguageServices.Workspace.<>c__DisplayClass43_0.<<OnEvaluationUpdateAsync>g__ProcessInitialEvaluationDataAsync|1>d.MoveNext()
--- End of stack trace from previous location ---
   at Microsoft.VisualStudio.ProjectSystem.VSCode.LanguageServices.Workspace.OnEvaluationUpdateAsync(IProjectVersionedValue`1 evaluationUpdate)
   at Microsoft.VisualStudio.ProjectSystem.VSCode.LanguageServices.Workspace.<>c__DisplayClass42_0.<<OnWorkspaceUpdateAsync>g__ApplyUpdateWithinLockAsync|1>d.MoveNext()
--- End of stack trace from previous location ---
   at Microsoft.VisualStudio.ProjectSystem.VSCode.LanguageServices.Workspace.<>c__DisplayClass42_0.<<OnWorkspaceUpdateAsync>g__ApplyUpdateWithinLockAsync|1>d.MoveNext()
--- End of stack trace from previous location ---
   at Microsoft.VisualStudio.ProjectSystem.VSCode.LanguageServices.Workspace.<>c__DisplayClass42_0.<<OnWorkspaceUpdateAsync>b__0>d.MoveNext()
--- End of stack trace from previous location ---
   at Microsoft.VisualStudio.Threading.ReentrantSemaphore.StackSemaphore.<>c__DisplayClass3_0.<<ExecuteAsync>b__0>d.MoveNext()
--- End of stack trace from previous location ---
```

```
System.AggregateException: Project system data flow 'Workspace update handler 27446469' closed because of an exception: The JSON-RPC connection with the remote party was lost before the request could complete.. (The JSON-RPC connection with the remote party was lost before the request could complete.)
---> (Inner Exception #0) StreamJsonRpc.ConnectionLostException: The JSON-RPC connection with the remote party was lost before the request could complete.
   at StreamJsonRpc.JsonRpc.InvokeCoreAsync(JsonRpcRequest request, Type expectedResultType, CancellationToken cancellationToken)
   at StreamJsonRpc.JsonRpc.InvokeCoreAsync[TResult](RequestId id, String targetName, IReadOnlyList`1 arguments, IReadOnlyList`1 positionalArgumentDeclaredTypes, IReadOnlyDictionary`2 namedArgumentDeclaredTypes, CancellationToken cancellationToken, Boolean isParameterObject)
   at Microsoft.VisualStudio.ProjectSystem.VSCode.LanguageServices.Handlers.ProjectFilePathAndDisplayNameEvaluationHandler.HandleAsync(IWorkspaceProject context, ProjectConfiguration projectConfiguration, IComparable version, IProjectChangeDescription projectChange, ContextState state, ITelemetryOperation telemetryOperation, IManagedProjectDiagnosticOutputService logger)
   at Microsoft.VisualStudio.ProjectSystem.VSCode.LanguageServices.Workspace.<>c__DisplayClass43_1.<<OnEvaluationUpdateAsync>g__ProcessProjectEvaluationHandlersAsync|5>d.MoveNext()
--- End of stack trace from previous location ---
   at Microsoft.VisualStudio.ProjectSystem.VSCode.LanguageServices.Workspace.<>c__DisplayClass43_0.<<OnEvaluationUpdateAsync>g__ApplyProjectEvaluationAsync|2>d.MoveNext()
--- End of stack trace from previous location ---
   at Microsoft.VisualStudio.ProjectSystem.VSCode.LanguageServices.Workspace.<>c__DisplayClass46_1`1.<<OnProjectChangedAsync>g__ApplyInBatchAsync|1>d.MoveNext()
--- End of stack trace from previous location ---
   at Microsoft.VisualStudio.Threading.ReentrantSemaphore.StackSemaphore.<>c__DisplayClass3_0.<<ExecuteAsync>b__0>d.MoveNext()
--- End of stack trace from previous location ---
   at Microsoft.VisualStudio.Threading.ReentrantSemaphore.StackSemaphore.ExecuteAsync(Func`1 operation, CancellationToken cancellationToken)
   at Microsoft.VisualStudio.ProjectSystem.OnceInitializedOnceDisposedUnderLockAsync.ExecuteUnderLockCoreAsync(Func`2 action, CancellationToken cancellationToken)
   at Microsoft.VisualStudio.ProjectSystem.VSCode.LanguageServices.Workspace.OnEvaluationUpdateAsync(IProjectVersionedValue`1 evaluationUpdate)
   at Microsoft.VisualStudio.ProjectSystem.VSCode.LanguageServices.Workspace.<>c__DisplayClass42_0.<<OnWorkspaceUpdateAsync>g__ApplyUpdateWithinLockAsync|1>d.MoveNext()
--- End of stack trace from previous location ---
   at Microsoft.VisualStudio.ProjectSystem.VSCode.LanguageServices.Workspace.<>c__DisplayClass42_0.<<OnWorkspaceUpdateAsync>g__ApplyUpdateWithinLockAsync|1>d.MoveNext()
--- End of stack trace from previous location ---
   at Microsoft.VisualStudio.ProjectSystem.VSCode.LanguageServices.Workspace.<>c__DisplayClass42_0.<<OnWorkspaceUpdateAsync>b__0>d.MoveNext()
--- End of stack trace from previous location ---
   at Microsoft.VisualStudio.Threading.ReentrantSemaphore.StackSemaphore.<>c__DisplayClass3_0.<<ExecuteAsync>b__0>d.MoveNext()
--- End of stack trace from previous location ---
   at Microsoft.VisualStudio.Threading.ReentrantSemaphore.StackSemaphore.ExecuteAsync(Func`1 operation, CancellationToken cancellationToken)
   at Microsoft.VisualStudio.ProjectSystem.OnceInitializedOnceDisposedUnderLockAsync.ExecuteUnderLockCoreAsync(Func`2 action, CancellationToken cancellationToken)
   at Microsoft.VisualStudio.ProjectSystem.DataReceivingBlockSlim`1.ProcessInputQueueAsync(InputDataBuffer inputQueue)
<--- (Inner Exception #0) 
```

### Null reference
I'm also seeing a fair number of null reference exceptions.  In one case I see this
```
System.AggregateException: One or more errors occurred. (Object reference not set to an instance of an object.)
---> (Inner Exception #0) System.NullReferenceException: Object reference not set to an instance of an object.
   at Microsoft.VisualStudio.ProjectSystem.Server.ProjectLaunchConfigurationService.<InitializeCoreAsync>g__ActionBlockCallback|25_0(IProjectVersionedValue`1 data)
   at Microsoft.VisualStudio.ProjectSystem.ActionBlockSlimSync`1.ProcessInputAsync(TInput input)
   at Microsoft.VisualStudio.ProjectSystem.DataReceivingBlockSlim`1.ProcessInputQueueAsync(InputDataBuffer inputQueue)
<--- (Inner Exception #0) 
```

the rest I mostly see
```
System.AggregateException: Project system data flow 'Workspace update handler 61249327' closed because of an exception: Object reference not set to an instance of an object.. (Object reference not set to an instance of an object.)
---> (Inner Exception #0) StreamJsonRpc.RemoteInvocationException: Object reference not set to an instance of an object.
   at StreamJsonRpc.JsonRpc.InvokeCoreAsync[TResult](RequestId id, String targetName, IReadOnlyList`1 arguments, IReadOnlyList`1 positionalArgumentDeclaredTypes, IReadOnlyDictionary`2 namedArgumentDeclaredTypes, CancellationToken cancellationToken, Boolean isParameterObject)
   at Microsoft.VisualStudio.ProjectSystem.VSCode.LanguageServices.Handlers.DynamicItemHandler.<>c__DisplayClass6_0.<<HandleProjectChangeAsync>g__ProcessAdditionsAsync|4>d.MoveNext()
--- End of stack trace from previous location ---
   at Microsoft.VisualStudio.ProjectSystem.VSCode.LanguageServices.Handlers.DynamicItemHandler.HandleProjectChangeAsync(IWorkspaceProject context, IComparable version, IProjectChangeDescription projectChange, ContextState state, ITelemetryOperation telemetryOperation, IManagedProjectDiagnosticOutputService logger)
   at Microsoft.VisualStudio.ProjectSystem.VSCode.LanguageServices.Handlers.DynamicItemHandler.HandleAsync(IWorkspaceProject context, IComparable version, IImmutableDictionary`2 projectChanges, ContextState state, ITelemetryOperation telemetryOperation, IManagedProjectDiagnosticOutputService logger)
   at Microsoft.VisualStudio.ProjectSystem.VSCode.LanguageServices.Workspace.<>c__DisplayClass43_1.<<OnEvaluationUpdateAsync>g__ProcessSourceItemsHandlersAsync|6>d.MoveNext()
--- End of stack trace from previous location ---
   at Microsoft.VisualStudio.ProjectSystem.VSCode.LanguageServices.Workspace.<>c__DisplayClass43_0.<<OnEvaluationUpdateAsync>g__ApplyProjectEvaluationAsync|2>d.MoveNext()
--- End of stack trace from previous location ---
   at Microsoft.VisualStudio.ProjectSystem.VSCode.LanguageServices.Workspace.<>c__DisplayClass46_1`1.<<OnProjectChangedAsync>g__ApplyInBatchAsync|1>d.MoveNext()
--- End of stack trace from previous location ---
   at Microsoft.VisualStudio.Threading.ReentrantSemaphore.StackSemaphore.<>c__DisplayClass3_0.<<ExecuteAsync>b__0>d.MoveNext()
--- End of stack trace from previous location ---
   at Microsoft.VisualStudio.Threading.ReentrantSemaphore.StackSemaphore.ExecuteAsync(Func`1 operation, CancellationToken cancellationToken)
   at Microsoft.VisualStudio.ProjectSystem.OnceInitializedOnceDisposedUnderLockAsync.ExecuteUnderLockCoreAsync(Func`2 action, CancellationToken cancellationToken)
   at Microsoft.VisualStudio.ProjectSystem.VSCode.LanguageServices.Workspace.OnEvaluationUpdateAsync(IProjectVersionedValue`1 evaluationUpdate)
   at Microsoft.VisualStudio.ProjectSystem.VSCode.LanguageServices.Workspace.<>c__DisplayClass42_0.<<OnWorkspaceUpdateAsync>g__ApplyUpdateWithinLockAsync|1>d.MoveNext()
--- End of stack trace from previous location ---
   at Microsoft.VisualStudio.ProjectSystem.VSCode.LanguageServices.Workspace.<>c__DisplayClass42_0.<<OnWorkspaceUpdateAsync>g__ApplyUpdateWithinLockAsync|1>d.MoveNext()
--- End of stack trace from previous location ---
   at Microsoft.VisualStudio.ProjectSystem.VSCode.LanguageServices.Workspace.<>c__DisplayClass42_0.<<OnWorkspaceUpdateAsync>b__0>d.MoveNext()
--- End of stack trace from previous location ---
   at Microsoft.VisualStudio.Threading.ReentrantSemaphore.StackSemaphore.<>c__DisplayClass3_0.<<ExecuteAsync>b__0>d.MoveNext()
--- End of stack trace from previous location ---
   at Microsoft.VisualStudio.Threading.ReentrantSemaphore.StackSemaphore.ExecuteAsync(Func`1 operation, CancellationToken cancellationToken)
   at Microsoft.VisualStudio.ProjectSystem.OnceInitializedOnceDisposedUnderLockAsync.ExecuteUnderLockCoreAsync(Func`2 action, CancellationToken cancellationToken)
   at Microsoft.VisualStudio.ProjectSystem.DataReceivingBlockSlim`1.ProcessInputQueueAsync(InputDataBuffer inputQueue)
<--- (Inner Exception #0) 
```

**Comment by NathanVG** [comment]
Everything was (kind of) working for me for a while but this error is popping up once again.

Honestly in combination with https://github.com/microsoft/vscode-dotnettools/issues/722 this is getting rediculous. It's nearly a year now that a microsoft plugin for a microsoft language in a microsoft editor has been close to unusable. 

```
=====================
10/2/2024 3:56:17 PM
LimitedFunctionality
System.AggregateException: Project system data flow 'Workspace update handler 48963094' closed because of an exception: Object reference not set to an instance of an object.. (Object reference not set to an instance of an object.)
---> (Inner Exception #0) StreamJsonRpc.RemoteInvocationException: Object reference not set to an instance of an object.
   at StreamJsonRpc.JsonRpc.InvokeCoreAsync[TResult](RequestId id, String targetName, IReadOnlyList`1 arguments, IReadOnlyList`1 positionalArgumentDeclaredTypes, IReadOnlyDictionary`2 namedArgumentDeclaredTypes, CancellationToken cancellationToken, Boolean isParameterObject)
   at Microsoft.VisualStudio.ProjectSystem.VSCode.LanguageServices.Handlers.DynamicItemHandler.<>c__DisplayClass6_0.<<HandleProjectChangeAsync>g__ProcessAdditionsAsync|4>d.MoveNext()
--- End of stack trace from previous location ---
   at Microsoft.VisualStudio.ProjectSystem.VSCode.LanguageServices.Handlers.DynamicItemHandler.HandleProjectChangeAsync(IWorkspaceProject context, IComparable version, IProjectChangeDescription projectChange, ContextState state, ITelemetryOperation telemetryOperation, IManagedProjectDiagnosticOutputService logger)
   at Microsoft.VisualStudio.ProjectSystem.VSCode.LanguageServices.Handlers.DynamicItemHandler.HandleAsync(IWorkspaceProject context, IComparable version, IImmutableDictionary`2 projectChanges, ContextState state, ITelemetryOperation telemetryOperation, IManagedProjectDiagnosticOutputService logger)
   at Microsoft.VisualStudio.ProjectSystem.VSCode.LanguageServices.Workspace.<>c__DisplayClass43_1.<<OnEvaluationUpdateAsync>g__ProcessSourceItemsHandlersAsync|6>d.MoveNext()
--- End of stack trace from previous location ---
   at Microsoft.VisualStudio.ProjectSystem.VSCode.LanguageServices.Workspace.<>c__DisplayClass43_0.<<OnEvaluationUpdateAsync>g__ApplyProjectEvaluationAsync|2>d.MoveNext()
--- End of stack trace from previous location ---
   at Microsoft.VisualStudio.ProjectSystem.VSCode.LanguageServices.Workspace.<>c__DisplayClass46_1`1.<<OnProjectChangedAsync>g__ApplyInBatchAsync|1>d.MoveNext()
--- End of stack trace from previous location ---
   at Microsoft.VisualStudio.Threading.ReentrantSemaphore.StackSemaphore.<>c__DisplayClass3_0.<<ExecuteAsync>b__0>d.MoveNext()
--- End of stack trace from previous location ---
   at Microsoft.VisualStudio.Threading.ReentrantSemaphore.StackSemaphore.ExecuteAsync(Func`1 operation, CancellationToken cancellationToken)
   at Microsoft.VisualStudio.ProjectSystem.OnceInitializedOnceDisposedUnderLockAsync.ExecuteUnderLockCoreAsync(Func`2 action, CancellationToken cancellationToken)
   at Microsoft.VisualStudio.ProjectSystem.VSCode.LanguageServices.Workspace.OnEvaluationUpdateAsync(IProjectVersionedValue`1 evaluationUpdate)
   at Microsoft.VisualStudio.ProjectSystem.VSCode.LanguageServices.Workspace.<>c__DisplayClass42_0.<<OnWorkspaceUpdateAsync>g__ApplyUpdateWithinLockAsync|1>d.MoveNext()
--- End of stack trace from previous location ---
   at Microsoft.VisualStudio.ProjectSystem.VSCode.LanguageServices.Workspace.<>c__DisplayClass42_0.<<OnWorkspaceUpdateAsync>g__ApplyUpdateWithinLockAsync|1>d.MoveNext()
--- End of stack trace from previous location ---
   at Microsoft.VisualStudio.ProjectSystem.VSCode.LanguageServices.Workspace.<>c__DisplayClass42_0.<<OnWorkspaceUpdateAsync>b__0>d.MoveNext()
--- End of stack trace from previous location ---
   at Microsoft.VisualStudio.Threading.ReentrantSemaphore.StackSemaphore.<>c__DisplayClass3_0.<<ExecuteAsync>b__0>d.MoveNext()
--- End of stack trace from previous location ---
   at Microsoft.VisualStudio.Threading.ReentrantSemaphore.StackSemaphore.ExecuteAsync(Func`1 operation, CancellationToken cancellationToken)
   at Microsoft.VisualStudio.ProjectSystem.OnceInitializedOnceDisposedUnderLockAsync.ExecuteUnderLockCoreAsync(Func`2 action, CancellationToken cancellationToken)
   at Microsoft.VisualStudio.ProjectSystem.DataReceivingBlockSlim`1.ProcessInputQueueAsync(InputDataBuffer inputQueue)
<--- (Inner Exception #0) 
===================

```

**Comment by dibarbet** [comment]
Suspecting an issue with Razor projects here.  If anyone has a project they can consistently reproduce it in and can share, please email me (dabarbet@microsoft.com).

We're working on some changes in parallel that will provide more details on what went wrong when it fails which should be in an upcoming prerelease.

**Comment by raurfang** [comment]
We are also hitting the issue for the past few weeks.
Narrowed it down to a workspace looking more or less like this:
```
Workspace root
- backend
  - dotnet-razor-project
- frontend
  - js-project-A
    - node_modules
  - js-project-B
    - node_modules
  - ...
- node_modules
```
The JS projects reference a lot of packages and have big `node_modules` folders. They are using `pnpm` for package management.

- Opening the  `<root>/backend` folder in VS Code does not trigger the issue.
- Opening the workspace `<root>` triggers the issue.
- Removing `/frontend/*/node_modules` folders stops the issue.
- Removing some `/frontend/*/node_modules` also stops the issue, but there is no rule which folders or files cause it

Rolling back the VS Code C# extension to version 2.41.26 or older also fixes the issue.

**Comment by ryzngard** [comment]
> We are also hitting the issue for the past few weeks. Narrowed it down to a workspace looking more or less like this:
> 
> ```
> Workspace root
> - backend
>   - dotnet-razor-project
> - frontend
>   - js-project-A
>     - node_modules
>   - js-project-B
>     - node_modules
>   - ...
> - node_modules
> ```
> 
> The JS projects reference a lot of packages and have big `node_modules` folders. They are using `pnpm` for package management.
> 
> * Opening the  `<root>/backend` folder in VS Code does not trigger the issue.
> * Opening the workspace `<root>` triggers the issue.
> * Removing `/frontend/*/node_modules` folders stops the issue.
> * Removing some `/frontend/*/node_modules` also stops the issue, but there is no rule which folders or files cause it
> 
> Rolling back the VS Code C# extension to version 2.41.26 or older also fixes the issue.

Thanks for this information. For the backend, does it include razor/cshtml files or just aspnetcore? If possible can you share a repro? If not possible to share, can you test with the vsix built in https://dev.azure.com/dnceng-public/cbb18261-c48f-4abb-8651-8cdcb5474649/_build/results?buildId=833689 ? That will help us know if the targeted fix spans to your scenario as well. Just looking at your information I would guess not, but I very much could be missing something.

**Comment by raurfang** [comment]
> > Rolling back the VS Code C# extension to version 2.41.26 or older also fixes the issue.
> 
> Thanks for this information. For the backend, does it include razor/cshtml files or just aspnetcore? If possible can you share a repro? If not possible to share, can you test with the vsix built in https://dev.azure.com/dnceng-public/cbb18261-c48f-4abb-8651-8cdcb5474649/_build/results?buildId=833689 ? That will help us know if the targeted fix spans to your scenario as well. Just looking at your information I would guess not, but I very much could be missing something.

Yes, the backend includes `razor` and `cshtml` files. I wasn't able to get reliable minimal reproduction steps, but the VSIX build seems to have helped!
It did take a lot of time to process the projects, but loaded them without the issue showing up, and all features seem to be working :+1:

**Comment by ryzngard** [comment]
Great, Thank you @raurfang! Hopefully we'll have this available in release soon

**Comment by dibarbet** [comment]
The fix should be available in either `2.50.27` (release) or `2.52.24` (prerelease)

**Comment by cklimowski1** [comment]
I'm still having this issue even with the new versions. Is there a ticket already tracking this? Making it very difficult to do work in .Net using VSCode, would love to see this resolved :)

**Comment by dibarbet** [comment]
@cklimowski1 would you mind filing a new issue and include the stack you're seeing?  We fixed a specific failure but there may be others.
